### PR TITLE
Tests for BatchDatasetOp

### PR DESCRIPTION
**Pull request by feihugis**
This PR enables AsGraphDefInternal to work with two versions of BatchDatasetOp and adds tests for `BatchDatasetOp`.

cc: @jsimsa

### CODE REVIEW

File: tensorflow/core/kernels/data/batch_dataset_op_test.cc [hunk 1]
```diff
@@ -0,0 +1,812 @@
...
+
+// Test Case 1: test BatchDataset v1 with a batch size that can evenly split the
+// input dataset.
+TestCase TestCase1() {
+  return {/*range_data_param*/ {0, 10, 1},
+          /*batch_size*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {2}),
+          /*drop_remainder*/
+          DatasetOpsTestBase::CreateTensor<bool>(TensorShape({}), {false}),
+          /*op_version*/ 1,
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/batch_dataset_op_test.cc]: you do not have to write for op_version 1 as it is not used anywhere

File: tensorflow/core/kernels/data/batch_dataset_op.cc [hunk 2]
```diff
@@ -114,8 +114,14 @@ class BatchDatasetOp : public UnaryDatasetOpKernel {
       TF_RETURN_IF_ERROR(b->AddScalar(batch_size_, &batch_size));
       Node* drop_remainder = nullptr;
       TF_RETURN_IF_ERROR(b->AddScalar(drop_remainder_, &drop_remainder));
-      TF_RETURN_IF_ERROR(b->AddDataset(
-          this, {input_graph_node, batch_size, drop_remainder}, output));
+      if (type_string() == "BatchDataset") {
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/batch_dataset_op.cc]: as per my comment in the test file, please undo this change

**Review by jsimsa — CHANGES_REQUESTED** [review]

**Comment by feihugis** [comment]
Got it. I rewrite this PR:

- https://github.com/tensorflow/tensorflow/pull/27876/commits/dac2bf1ba9361da4c416776fece91564f7bc174f: Minor updates of the comments in dataset.h file

- https://github.com/tensorflow/tensorflow/pull/27876/commits/bed70239456192e2aa7da32cc34c966c40914af4: add the tests for BatchDatasetOp (remove the parts for op_version 1)

**Review by jsimsa — DISMISSED** [review]

**Comment by feihugis** [comment]
The failure log for the test `Ubuntu CC` can't be accessed. Could you please help check the log?

**Comment by feihugis** [comment]
The `Ubuntu CC` test runs successfully on my laptop. @gbaned Could you please help re-trigger the test?

**Comment by feihugis** [comment]
Thanks @gbaned for re-triggering the tests. 

Recently, some changes (https://github.com/tensorflow/tensorflow/commit/0c23c60a1a013901c85e340c60afdfd9ec6e9dce#diff-79a93a3e76fded5f3522ad99b27976fd) were added to `BatchDatasetV2`, which causes the test failures. Will revise this PR accordingly.

**Comment by gbaned** [comment]
@feihugis Any update please?  Thanks!

**Comment by feihugis** [comment]
@gbaned Sorry for the delay as I'm in a conference these two days. Will fix the issue today.

**Comment by feihugis** [comment]
> Recently, some changes ([0c23c60#diff-79a93a3e76fded5f3522ad99b27976fd](https://github.com/tensorflow/tensorflow/commit/0c23c60a1a013901c85e340c60afdfd9ec6e9dce#diff-79a93a3e76fded5f3522ad99b27976fd)) were added to `BatchDatasetV2`, which causes the test failures. Will revise this PR accordingly.

@jsimsa @gbaned The new attr `parallel_copy` is added to `BatchDatasetOpTest` via https://github.com/tensorflow/tensorflow/pull/27876/commits/6e7399a4ca8df3b7d55f981605d32a00aaf51890. Could you please have a look at the change when you get a chance?

**Comment by feihugis** [comment]
> Recently, some changes ([0c23c60#diff-79a93a3e76fded5f3522ad99b27976fd](https://github.com/tensorflow/tensorflow/commit/0c23c60a1a013901c85e340c60afdfd9ec6e9dce#diff-79a93a3e76fded5f3522ad99b27976fd)) were added to `BatchDatasetV2`, which causes the test failures. Will revise this PR accordingly.
> 
@jsimsa `BatchDatasetV2` got a new attribute (`parallel_copy`), so I revise this PR accordingly by adding `parallel_copy` to `BatchDatasetOpTest` via [6e7399a](https://github.com/tensorflow/tensorflow/commit/6e7399a4ca8df3b7d55f981605d32a00aaf51890). Could you please review the change when you have time?

**Review by jsimsa — APPROVED** [review]

**Pull request merged**